Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -69,6 +71,18 @@ extension PaymentRequestExtension: TabContentScript {
                 return
             }
             
+            // All currencies should match
+            guard body.details.displayItems.map({$0.amount.currency}).allSatisfy({ $0 == body.details.total.amount.currency }) else {
+                sendPaymentRequestError(errorName: PaymentRequestErrors.TypeError.rawValue, errorMessage: Strings.invalidDetailsMessage)
+                return
+            }
+            
+            // Sum of individual items does not match the total
+            guard Int(body.details.total.amount.value) == body.details.displayItems.map({(Int($0.amount.value) ?? 0 )}).reduce(0, +) else {
```

Comment:
You can use `compactMap` here instead of `map` instead of `?? 0`

I.e. `body.details.displayItems.compactMap({ Int($0.amount.value) }).reduce(0, +)`